src/main/java/core/dao/manufacturer/ManufacturerDaoPostgres.java

```diff
@@ -0,0 +1,140 @@
+            getById.close();
+            
+            List<Manufacturer> fromResultSet = getFromResultSet(resultSet);
+            if (fromResultSet.size() == 1) {
+                return Optional.ofNullable(fromResultSet.get(0));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Fail to get manufacturer by id " + id, e);
+        }
+    }
+    
+    @Override
+    public List<Manufacturer> getAll() {
+        String select = "SELECT id, name, country"
+                        + " FROM manufacturers"
+                        + " WHERE deleted = false;";
+        try (Connection con = ConnectionUtils.getConnection()) {
+            PreparedStatement getAll = con.prepareStatement(select);
+            ResultSet resultSet = getAll.executeQuery();
+            getAll.close();
+    
+            return getFromResultSet(resultSet);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get all manufacturers from database", e);
+        }
+    }
+    
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String update = "UPDATE manufacturers SET name = ?,"
+                        + " country = ? WHERE id = ? AND deleted = false";
+        try (Connection con = ConnectionUtils.getConnection()) {
+            PreparedStatement updateStatement = con.prepareStatement(update);
+            updateStatement.setString(1, manufacturer.getName());
+            updateStatement.setString(2, manufacturer.getCountry());
+            updateStatement.setLong(3, manufacturer.getId());
+            updateStatement.executeUpdate();
+            updateStatement.close();
+        } catch (SQLException exception) {
+            throw new DataProcessingException(
+                    String.format("Failed to update the %s",
+                            manufacturer.toString()
+                    ),
+                    exception);
+        }
+        return manufacturer;
+    }
+    
+    @Override
+    public boolean delete(Long id) {
+        String delete = "UPDATE manufacturers SET deleted = true WHERE id = ?";
+        int updated;
+        try (Connection con = ConnectionUtils.getConnection()) {
+            PreparedStatement deleteStatement = con.prepareStatement(delete);
+            deleteStatement.setLong(1, id);
+            updated = deleteStatement.executeUpdate();
+            deleteStatement.close();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to delete the manufacturer at id = " + id, e);
+        }
+        return updated > 0;
+    }
+    
+    @Override
+    public boolean delete(Manufacturer manufacturer) {
+        return delete(manufacturer.getId());
+    }
+    
+    private List<Manufacturer> getFromResultSet(ResultSet resultSet) {
+        List<Manufacturer> list = new ArrayList<>();
+        try {
```

Comment:
We are handling `SQLException` in `get()` and `getAll()` anyway, so it may be better to add this exception to the method signature to reduce code duplication

Reply:
Nice, thx